src/json/json_schema.rs

```diff
@@ -34,7 +35,11 @@ impl Default for JsonSchema {
 
 impl JsonSchema {
   // TODO: validate `JsonLike` instead of fixing on `async_graphql::Value`
-  pub fn validate(&self, value: &async_graphql::Value) -> Valid<(), &'static str> {
+  pub fn validate(
+    &self,
+    value: &async_graphql::Value,
+    schema_map: &HashMap<String, JsonSchema>,
+  ) -> Valid<(), &'static str> {
```

Comment:
I think it would be better to match on `value` instead of `self`.

Reply:
Hi @tusharmath, I had a similar thought but did not proceed in that direction assuming the following: For most cases `Schema Definition` is a subset of `API response` i.e response will contain way more fields than defined in the schema. In that case, it might be better to check if the defined fields satisfy the schema and leave the rest untouched? I am not very familiar with the usage patterns of graphQL, so please correct me if that's not the case and I will adjust it accordingly. Thanks.

Reply:
It could happen both ways. The types could have more fields than the actual value. I think matching over the value will simplify the logic quite a lot and drop the usage of the HashMap

Reply:
Ah, ok. Sure. I will try that approach.